modules/dnn/misc/python/test/test_dnn.py

```diff
@@ -21,15 +21,11 @@ def box2str(box):
     width, height = box[2] - left, box[3] - top
     return '[%f x %f from (%f, %f)]' % (width, height, left, top)
 
-def normAssertDetections(test, ref, out, confThreshold=0.0, scores_diff=1e-5, boxes_iou_diff=1e-4):
-    ref = np.array(ref, np.float32)
-    refClassIds, testClassIds = ref[:, 1], out[:, 1]
-    refScores, testScores = ref[:, 2], out[:, 2]
-    refBoxes, testBoxes = ref[:, 3:], out[:, 3:]
-
+def normAssertDetections(test, refClassIds, refScores, refBoxes, testClassIds, testScores, testBoxes,
+                 confThreshold=0.0, scores_diff=1e-5, boxes_iou_diff=1e-4):
     matchedRefBoxes = [False] * len(refBoxes)
     errMsg = ''
-    for i in range(len(refBoxes)):
+    for i in range(len(testBoxes)):
```

Comment:
Why we not iterate over reference boxes?

Reply:
if len(testBoxes) < len(refBoxes) we get `IndexError`

Reply:
Oh, sorry. I agree with this change. In the origin C++ implementation it aslo iterates over test boxes: https://github.com/opencv/opencv/blob/master/modules/dnn/test/test_common.impl.hpp